src/topology/separation.lean

```diff
@@ -16,6 +16,46 @@ variables {α : Type u} {β : Type v} [topological_space α]
 
 section separation
 
+/--
+`separate` is a predicate on pairs of `finset`s of a topological space.  It holds if the two
+`finset`s are contained in disjoint open sets.
+-/
+def separate : finset α → finset α → Prop :=
+  λ (s t : finset α), disjoint s t → ∃ U V : (set α), (is_open U) ∧ is_open V ∧
+  (∀ a : α, a ∈ s → a ∈ U) ∧ (∀ a : α, a ∈ t → a ∈ V) ∧ disjoint U V
+
+lemma separate_symm (s t : finset α) : separate s t → separate t s :=
+begin
+ intros h1 d,
+ obtain ⟨U, V, oU, oV, aU, bV, UV⟩ := h1 (disjoint.symm d),
+ exact ⟨V, U, oV, oU, bV, aU, disjoint.symm UV⟩
+end
+
```

Comment:
Please also add `separate_comm`, which is the `iff` variant of `symm`. That's useful for rewriting.

Reply:
Added!  I also called it `separate.comm` instead of `separate_comm`: I hope that this is right!

Reply:
Putting a lemma in the `separate` namespace is useful when the lemma takes an argument of type `separate _ _` as then we can use "dot notation": given a hypothesis of the form `h : separate _ _`, we could then write `h.symm`, for example.

So lemmas like `separate.comm` and `separate_empty_right` which don't take such arguments won't really benefit from the dot notation.

Reply:
@bryangingechen Thanks you also for the explanation!  I ended up putting the `.`: even though it might not benefit from projection, would it mean that if I `open separate` then I can use them without typing `separate.`, at least some of the times?

Reply:
@adomani yes, that would also work, although Lean might complain that it couldn't figure out whether you mean `separate.comm` or `foobar.comm` (if that exists, and you have `open foobar` in place).